components/bin/makeAll

```diff
@@ -0,0 +1,134 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Processes all the build and webpack files in a directory
+ *                  or collection of directories
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+
+const fs = require('fs');
+const path = require('path');
+const {execSync} = require('child_process');
+
+/*
+ * Get the directories to process
+ */
+const dirs = process.argv.slice(2);
+if (dirs.length === 0) {
+    dirs.push('.');
+}
+
+/**
+ * The path to the bin/build executable
+ */
+const build = path.join(__dirname, 'build');
+
+/**
+ * Regular expressions for the components directory and the mathjax3 .js location
+ */
+const compRE = new RegExp(path.dirname(__dirname).replace(/([\\.{}[\]()?*^$])/g, '\$1'), 'g');
+const rootRE = new RegExp(path.join(path.dirname(path.dirname(__dirname)), 'mathjax3')
+                          .replace(/([\\.{}[\]()?*^$])/g, '\$1'), 'g');
+
+/**
+ * Process the contents of an array of directories
+ *
+ * @param {string} dirs   The directories to process
+ */
+function processList(dirs) {
+    for (const dir of dirs) {
+        const fulldir = path.resolve(dir);
+        processDir(fulldir, buildLib);
+        processDir(fulldir, webpackLib);
+    }
+}
+
+/**
+ * Run an action (build or webpack) on a directory and its subdirectories
+ *
+ * @param {string} dir        The directory to process
+ * @param {Function} action   The action to take
+ */
+function processDir(dir, action) {
+    action(dir);
+    processSubdirs(dir, action);
+}
+
+/**
+ * Look for subdirectories and process them
+ *
+ * @param {string} dir        The directory whose subdirectories are to be processed
+ * @param {Function} action   The action to take
+ */
+function processSubdirs(dir, action) {
+    for (const name of fs.readdirSync(dir)) {
+        const file = path.join(dir, name);
+        if (fs.lstatSync(file).isDirectory()) {
+            processDir(file, action);
+        }
+    }
+}
+
+/**
+ * Run bin/build if there is a configuration file for it
+ *
+ * @param {string} dir   The directory to check
+ */
+function buildLib(dir) {
+    const file = path.join(dir, 'build.json');
+    if (!fs.existsSync(file)) return;
+    console.log('Building ' + dir.replace(compRE, ''));
+    const wd = process.cwd();
+    try {
+        process.chdir(dir);
+        const result = execSync(build);
+        console.log('    ' + String(result).replace(/\n/g, '\n    '));
```

Comment:
Indentation?

Reply:
This is just for console output to tell you what is happening.  I don't think that needs to be configurable.